peepingtom/utils/dynamo_helper.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+import pandas as pd
+from eulerangles import euler2matrix
+
+from .validators import columns_in_df
+from .constants.dynamo_constants import dynamo_table_coordinate_headings, dynamo_table_shift_headings, \
+    dynamo_euler_angle_headings
+
+
+def df_to_xyz(df: pd.DataFrame):
+    """
+
+    Parameters
+    ----------
+    df : dynamo format table file as DataFrame (usually the result of dynamotable.read)
+
+    Returns (n, 3) ndarray of xyz positions from the DataFrame
+    -------
+
+    """
+    assert columns_in_df(dynamo_table_coordinate_headings + dynamo_table_shift_headings, df)
+    xyz = df[dynamo_table_coordinate_headings] + df[dynamo_table_shift_headings]
+    return xyz.to_numpy()
+
+
+def df_to_euler_angles(df: pd.DataFrame):
+    """
+
+    Parameters
+    ----------
+    df : dynamo format table file as DataFrame (usually the result of dynamotable.read)
+
+    Returns (n, 3) ndarray of euler angles (tdrot, tilt, narot) from dataframe
+    -------
+
+    """
+    assert columns_in_df(dynamo_euler_angle_headings)
```

Comment:
Here and in a few other places you used `assert` for non-test-related things. It shouldn't be used for non-debugging purposes, because reasons (https://stackoverflow.com/a/54252850), so here I would replace it with a raise. We should probably have a specific exception, since we may want to handle it higher up somehow.
```
if not ...:
    raise ...
```
or, elsewhere:
```
if not isinstance(a, b):
    raise ...
```